app/com/linkedin/drelephant/spark/fetchers/SparkFetcher.scala

```diff
@@ -53,23 +53,26 @@ class SparkFetcher(fetcherConfigurationData: FetcherConfigurationData)
     sparkConf
   }
 
-  private[fetchers] lazy val sparkRestClient: SparkRestClient = new SparkRestClient(sparkConf)
-
-  private[fetchers] lazy val sparkLogClient: Option[SparkLogClient] = {
+  private[fetchers] lazy val eventLogSource: EventLogSource = {
     val eventLogEnabled = sparkConf.getBoolean(SPARK_EVENT_LOG_ENABLED_KEY, false)
-    if (eventLogEnabled) Some(new SparkLogClient(hadoopConfiguration, sparkConf)) else None
+    val useRestForLogs = Option(fetcherConfigurationData.getParamMap.get("use_rest_for_eventlogs"))
+      .exists(_.toBoolean)
+    if (!eventLogEnabled) {
+      EventLogSource.None
+    } else if (useRestForLogs) EventLogSource.Rest else EventLogSource.WebHdfs
   }
 
-  private[fetchers] lazy val useRestForLogs: Boolean = {
-    Option(fetcherConfigurationData.getParamMap.get("use_rest_for_eventlogs"))
-      .exists(_.toBoolean)
+  private[fetchers] lazy val sparkRestClient: SparkRestClient = new SparkRestClient(sparkConf)
+
+  private[fetchers] lazy val sparkLogClient: SparkLogClient = {
```

Comment:
I've deliberately removed `Option[_]` from the type, because otherwise we allow illegal state: `sparkLogClient.isDefined` and `eventLogSource == None`.